lib/bundler/cli/gem.rb

```diff
@@ -97,10 +98,14 @@ def run
             "test/test_helper.rb.tt" => "test/test_helper.rb",
             "test/newgem_test.rb.tt" => "test/#{namespaced_path}_test.rb"
           )
+        when "test-unit"
+          templates.merge!(
+            "test/test_newgem.rb.tt" => "test/test_#{namespaced_path}.rb"
+          )
         end
       end
 
-      config[:test_task] = config[:test] == "minitest" ? "test" : "spec"
+      config[:test_task] = config[:test] == "minitest" || config[:test] == "test-unit" ? "test" : "spec"
```

Comment:
`config[:test] == "rspec" ? "spec" : "test"` will be simpler.
See also: https://github.com/bundler/bundler/pull/5521/files#diff-2ee2e162f4eceee05f93f6197078d10bR99